### DISCUSSION TITLE
How to apply k-fold cross-validation in yolov5？

### DISCUSSION THREAD

**Discussion by Chenzkun** [ultralytics/yolov5#9245] [Q&A]
hello@glenn-jocher，
I want to use k-fold cross-validation in yolov5, but I don't know how to do it

**Comment by spacewalk01** [comment]
https://www.kaggle.com/code/ayuraj/train-yolov5-cross-validation-ensemble-w-b

**Comment by spacewalk01** [comment]
Split your data into k-folds as follows and train a model on each dataset 
```
# Create .yaml file 
import yaml

for fold in range(NUM_FOLD):
    data_yaml = dict(
        train = f'../dataset_folds_{fold}/images/train',
        val = f'../dataset_folds_{fold}/images/valid',
        nc = 2,
        names = ['none', 'opacity']
    )

    # Note that I am creating the file in the yolov5/data/ directory.
    with open(f'yolov5/data/data_fold_{fold}.yaml', 'w') as outfile:
        yaml.dump(data_yaml, outfile, default_flow_style=True)
    
%cat yolov5/data/data_fold_0.yaml
```

**Reply by Chenzkun** [reply]
Is this to create a new yaml file to replace coco128.yaml in the source code?

**Reply by spacewalk01** [reply]
Yes

**Reply by Chenzkun** [reply]
But you've only created a few yaml files, it doesn't say how to train them

**Comment by TonSarun** [comment]
https://docs.ultralytics.com/guides/kfold-cross-validation/#setup
